daemon/cluster/noderunner.go

```diff
@@ -166,6 +168,46 @@ func (n *nodeRunner) handleControlSocketChange(ctx context.Context, node *swarmn
 	}
 }
 
+func (n *nodeRunner) watchClusterEvents(ctx context.Context, conn *grpc.ClientConn) {
+	if n.cluster == nil {
+		return
+	}
```

Comment:
I can't see any code path where it is possible for `n.cluster` to be nil. I'd suggest removing this - it is a little misleading to suggest that `n.cluster` can be nil, when other code would blow up in this case.